### ISSUE TITLE
[TF2] shavit-misc's NoWeaponDrops is broken

### ISSUE THREAD

**Issue by jedso** [shavitush/bhoptimer#1159]
You can't purposefully drop items in TF2 like you can in CS:S/GO (no `drop` command). However, when you die (e.g. `kill`), your current weapon is dropped for other team members to use along with an ammo pack. The big difference in TF2 though, is that this dropped weapon becomes a new **tf_dropped_weapon** entity and therefore never satisfies the `FindEntityByClassname(ent, "weapon_*")` check in shavit-misc's Timer_Cron when gCV_NoWeaponDrops.BoolValue is true.
https://github.com/shavitush/bhoptimer/blob/68a57b912341b9f02e8ce3b48b05773b58771d8c/addons/sourcemod/scripting/shavit-misc.sp#L959-L970

SDKHook_WeaponDrop also doesn't behave as you would expect in TF2. When you die, the WeaponDrop callback contains the entity index of the last equipped weapon but NOT the entity index of the newly created **tf_dropped_weapon**. Unless the player changes their loadout, the weapon entity index remains the same before and after respawning, so the only thing  OnWeaponDrop does is kill whatever weapon entity was equipped at time of death.
https://github.com/shavitush/bhoptimer/blob/68a57b912341b9f02e8ce3b48b05773b58771d8c/addons/sourcemod/scripting/shavit-misc.sp#L1474-L1480

So basically neither Timer_Cron or the WeaponDrop callback kill dropped weapons in TF2. Luckily though TF2 has a built-in ConVar for this functionality, ` tf_dropped_weapon_lifetime`, so there’s no need to use a timer or hook WeaponDrop.
